api/clients/v2/verification/cert_verifier_generic.go

```diff
@@ -0,0 +1,354 @@
+package verification
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/coretypes"
+	disperser "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common"
+	generic_verifier_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifierV3"
+	cert_type_binding "github.com/Layr-Labs/eigenda/contracts/bindings/IEigenDACertTypeBindings"
+	opsr_binding "github.com/Layr-Labs/eigenda/contracts/bindings/OperatorStateRetriever"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// GenericCertVerifier is responsible for making eth calls against version agnostic CertVerifier contracts to ensure
+// cryptographic and structural integrity of EigenDA certificate types.
+// The V3 cert verifier contract is located at:
+// https://github.com/Layr-Labs/eigenda/blob/master/contracts/src/periphery/cert/v3/EigenDACertVerifierV3.sol
+type GenericCertVerifier struct {
```

Comment:
Again here, why create a new struct, instead of just modifying the old struct?

Reply:
better future proofing for inevitable code deletion

Reply:
I think the correct solution here is to modify the existing structure, instead of duplicating code

Reply:
Ok, I think I might be convinced that separate is the way to go.

One thing that could help might be naming this "new" cert verifier the same as the original one. i.e. call it `CertVerifier`, instead of `GenericCertVerifier`

Since you've already dubbed the old logic to be `legacy`, the unqualified `CertVerifier` name is wide open. `Generic` won't have much meaning, once the `legacy` version has been deleted. 

This rename would help git diffs properly show that this `generic` implementation is the spiritual successor to the original implementation, and generate a diff which is easier to grasp

Reply:
0d0f7bbd771bccdab89ebb767b78dcd2163673bb